status/reporters.go

```diff
@@ -413,3 +489,134 @@ func (r *GitHubReporter) ReportStatus(k8sClient client.Client, ctx context.Conte
 
 	return nil
 }
+
+// ReportStatusForSnapshot creates CheckRuns when using GitHub App integration.
+// When using GitHub webhook integration a commit status
+func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot, details *[]gitops.IntegrationTestStatusDetail) error {
```

Comment:
This function is incredibly long with so huge ratio of duplicated code, it needs refactor

Reply:
I feel the same, let me update it. Thanks

Reply:
fixed.